### PR TITLE
Remove what it is header from all En docs

### LINKED ISSUE #2575: Remove the What it is Header completely
Remove the what it is header from templates and all terms.

### PR DESCRIPTION
**Pull request by seokho-son**
### Describe your changes
fix #2575 

### Related issue number or link (ex: `resolves #issue-number`)
#2575 
#2242

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

**Comment by seokho-son** [comment]
preview example. 
https://deploy-preview-2652--cncfglossary.netlify.app/api-gateway/

File: content/en/_TEMPLATE.md [hunk 1]
```diff
@@ -4,9 +4,8 @@ status: Feedback Appreciated
 category: concept
 ---
 
-## What it is
 
-Quick summary of the concept and what it is.
```
Comment (nate-double-u) [inline content/en/_TEMPLATE.md:9]: We can probably leave the "and what it is" line here as this is a section that is now the introduction.

File: content/en/style-guide/_index.md [hunk 2]
```diff
@@ -48,7 +48,6 @@ status:
 category: 
 ---
 
-## What it is
```
Comment (nate-double-u) [inline content/en/style-guide/_index.md:51]: let's remove one more blank line here too

**Review by nate-double-u — APPROVED** [review]
Thanks so much for doing this @seokho-son! I have a couple of comments, but don't let these be blockers if you'd like to merge as is.

**Comment by seokho-son** [comment]
@nate-double-u Thanks for the review :)
I applied all your suggestions. ;)

**Review by iamNoah1 — APPROVED** [review]
lgtm and thanks a bunch for the efforts

**Pull request merged**